app/models/user.rb

```diff
@@ -548,6 +548,9 @@ def update_pseud_name
           # shouldn't be able to get here, but just in case
           Pseud.create!(name: login, user_id: id)
         end
+      else
+        new_pseud.name = login
+        new_pseud.save!(validate: false)
```

Comment:
A lot of the branches in this function are very similar (setting `pseud.name = login` and then calling `pseud.save!`). The choice of when to call `save!(validate: false)` vs `save!` seems mostly arbitrary, so the only thing the branches are doing is determining which pseud to modify. Would it be possible to refactor the whole method so that `pseud.name = login` and `pseud.save!` are called exactly once at the end?